src/pages/blogDetailPage/BlogDetail.jsx

```diff
@@ -0,0 +1,82 @@
+import {useEffect, useState} from "react";
+import { Link, useParams, useNavigate } from "react-router-dom";
+import axios from "axios";
+import { CaretLeft, Clock } from "@phosphor-icons/react";
+import Button from "../../components/button/Button.jsx";
+import ErrorMessage from "../../components/errorMessage/ErrorMessage.jsx";
+import formatDateString from "../../helpers/formatDateString.js";
+import './BlogDetail.css';
+
+
+function BlogDetail() {
+    const { id } = useParams();
+    const navigate = useNavigate();
+    const [postDetails, setPostDetails] = useState({});
+    const [deleteMessage, setDeleteMessage] = useState('');
+    const [error, toggleError] = useState(false);
+    const [loading, toggleLoading] = useState(false);
+
+    useEffect(() => {
+        async function fetchPostDetails() {
+            toggleError(false);
+            toggleLoading(true);
+
+            try {
+                const response = await axios.get(`http://localhost:3000/posts/${id}`);
+                console.log(response.data);
+                setPostDetails(response.data);
+            } catch (e) {
+                console.error(e);
+                toggleError(true);
+            } finally {
+                toggleLoading(false);
+            }
+        }
+
+        fetchPostDetails();
+    }, []);
+
+    async function deletePost() {
+        try {
+            const response = await axios.delete(`http://localhost:3000/posts/${id}`);
+            console.log(response.data);
+            setDeleteMessage("De post is succesvol verwijderd!🎉");
+
+            setTimeout(() => { navigate("/posts") }, 1000);
+
+        } catch (e) {
+            console.error(e);
+            setDeleteMessage("Het verwijderen is niet gelukt.😿 Probeer het opnieuw.");
+        }
+    }
+
+    return (
+        <section className="outer-content-container">
+            <div className="blog-detail--container">
+                {Object.keys(postDetails).length > 0 ?
```

Comment:
Hier wordt de check dan wat simpeler
```suggestion
                {postDetails ?
```